go/vt/vtgate/planbuilder/operators/apply_join.go

```diff
@@ -73,3 +75,56 @@ func (a *ApplyJoin) Clone(inputs []Operator) Operator {
 func (a *ApplyJoin) Inputs() []Operator {
 	return []Operator{a.LHS, a.RHS}
 }
+
+var _ joinOperator = (*ApplyJoin)(nil)
+
+func (a *ApplyJoin) tableID() semantics.TableSet {
+	return TableID(a)
+}
+
+func (a *ApplyJoin) getLHS() Operator {
+	return a.LHS
+}
+
+func (a *ApplyJoin) getRHS() Operator {
+	return a.RHS
+}
+
+func (a *ApplyJoin) setLHS(operator Operator) {
+	a.LHS = operator
+}
+
+func (a *ApplyJoin) setRHS(operator Operator) {
+	a.RHS = operator
+}
+
+func (a *ApplyJoin) makeInner() {
+	a.LeftJoin = false
+}
+
+func (a *ApplyJoin) isInner() bool {
+	return !a.LeftJoin
+}
+
+func (a *ApplyJoin) addJoinPredicate(ctx *plancontext.PlanningContext, expr sqlparser.Expr) error {
+	bvName, cols, predicate, err := BreakExpressionInLHSandRHS(ctx, expr, TableID(a))
+	if err != nil {
+		return err
+	}
+	lhs, idxs, err := PushOutputColumns(ctx, a.LHS, cols...)
+	if err != nil {
+		return err
+	}
+	a.LHS = lhs
+	vars := map[string]int{}
+	for i, idx := range idxs {
+		vars[bvName[i]] = idx
+	}
+	rhs, err := PushPredicate(ctx, predicate, a.RHS)
+	if err != nil {
+		return err
+	}
+	a.RHS = rhs
```

Comment:
you could use `a.setRHS(rhs)` here

Reply:
sure. what would be the benefit? the interface methods are useful when you can't touch the fields directly, but here I can